src/Client/StatsD.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace HelloFresh\Stats\Client;
+
+
+use HelloFresh\Stats\Client;
+use HelloFresh\Stats\HTTPMetricAlterCallback;
+use HelloFresh\Stats\Incrementer;
+use HelloFresh\Stats\State;
+use HelloFresh\Stats\Timer\StatsD as StatsDTimer;
+use League\StatsD\Client as StatsDClient;
+
+class StatsD extends Base implements Client
+{
+    /** @var string */
+    protected $httpRequestSection;
+    /** @var HTTPMetricAlterCallback */
+    protected $httpMetricAlterCallback;
+    /** @var StatsDClient */
+    protected $client;
+
+    /**
+     * StatsD constructor.
+     *
+     * @param string $dsn statsd connection dsn
+     */
+    public function __construct($dsn)
+    {
+        $url = parse_url($dsn);
+
+        $params = parse_str(empty($url['query']) ? '' : $url['query']);
+        $options = [
+            'host' => $url['host'],
+            'port' => $url['port'],
+            'namespace' => empty($params['ns']) ? '' : $params['ns'],
+            'timeout' => empty($params['timeout']) ? null : (float)$params['timeout'],
+            'throwConnectionExceptions' => empty($params['error']) ? true : (bool)$params['error'],
```

Comment:
Instead of all the empty check you could do:
```PHP
$urlInfo = parse_url($dsn);
$options = array_merge(
['ns' => '', 'timeout' => null, 'error' => true],
$urlInfo
);
```